boards/arm/a1x/pcduino-a10/configs/nsh/defconfig

```diff
@@ -46,3 +46,4 @@ CONFIG_START_YEAR=2013
 CONFIG_SYMTAB_ORDEREDBYNAME=y
 CONFIG_SYSTEM_NSH=y
 CONFIG_UART0_SERIAL_CONSOLE=y
+CONFIG_ARCH_BOARD_COMMON=y
```

Comment:
not all board support common folder, it is wrong to add CONFIG_ARCH_BOARD_COMMON into ALL defconfig.

Reply:
I still need to test, but if we use the exported archive (from make export) to build apps we have the error where it says cannot fine `libboard.a`. I am not sure it is true for all boards though soo I will test have to test that out.

Reply:
hey @xiaoxiang781216, i was thinking of addressing this PR finally. I wanted your input regarding this discussing. Do you thins adding `CONFIG_ARCH_BOARD_COMMON=y` to boards that support common folder is acceptable? For now i was thinking of adding it to the STM32 boards and XMC4, or is it generally not a good idea and we want the user to be responsible for setting this option?